dev-lua/sol2/sol2-3.2.2.ebuild

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+LUA_COMPAT=( lua5-{1,3,4} luajit )
+inherit cmake lua
+
+DESCRIPTION="Header-only C++ <-> Lua API wrapper"
+HOMEPAGE="https://github.com/ThePhD/sol2"
+SRC_URI="https://github.com/ThePhD/sol2/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+REQUIRED_USE="${LUA_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+RDEPEND="${LUA_DEPS}"
+DEPEND="
+	test? (
+		dev-cpp/catch:1
+		${LUA_DEPS}
+	)
+"
+BDEPEND="
+	test? (
+		virtual/pkgconfig
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-luajit-pkgconf.patch
+	"${FILESDIR}"/${P}-catch-depend.patch
+)
+
+src_prepare() {
+	# TODO: remove this sed on next bump
+	sed -i -e 's/-Werror//' \
+		tests/runtime_tests/CMakeLists.txt \
+		tests/regression_tests/1011/CMakeLists.txt \
+		tests/config_tests/function_pointers/CMakeLists.txt \
+		examples/customization/CMakeLists.txt \
+		examples/CMakeLists.txt || die
+
+	cmake_src_prepare
+}
+
+src_configure() {
+	cmake_configure_wrapper() {
+		local mycmakeargs=(
+			-DSOL2_BUILD_LUA=no
+			-DSOL2_TESTS="$(usex test)"
+			-DSOL2_LUA_VERSION="${ELUA}"
+			-DCATCH_INC_DIR="${ESYSROOT}/usr/include/catch"
+		)
+		cmake_src_configure
+	}
+	lua_foreach_impl cmake_configure_wrapper
+}
+
+src_compile() {
+	lua_foreach_impl cmake_src_compile
+}
+
+src_test() {
+	test_wrapper() {
+		if [[ "${ELUA}" == luajit ]]; then
+			echo "Skipping test due to https://github.com/ThePhD/sol2/issues/1221"
+		else
+			cmake_src_test
+		fi
+	}
+	lua_foreach_impl test_wrapper
+}
+
+src_install() {
+	lua_foreach_impl cmake_src_install
```

Comment:
One more thing...

Haven't tried but is foreach needed here? Considering this doesn't actually install any lua files, it's C++ headers.

It's actually making the whole LUA_TARGETS thing a bit awkward outside of tests. If anything I feel like lua targets (required_use, lua deps, etc..) could be made optional behind `test?`. Not that I really explored the idea.

Reply:
Perhaps the category is even wrong, dev-cpp may make more sense.

Reply:
> Haven't tried but is foreach needed here?

cmake_src_install doesn't work by itself, because the `_build` directory is empty - cmake was never run there.

> could be made optional behind test?

That would require something like...
```
if test; then
    lua_foreach_impl cmake_src_install
else
    cmake_src_install
fi
```
which looks very strange

I would really like review from marecki@. Same for "Perhaps the category is even wrong"

Reply:
At this point I can't see this as a dev-lua ebuild at all, it's a dev-cpp headers package with lua-based tests.

Reply:
I'd just be about separating the two, this include compile-time so use test forearch should avoidable.